ena_portal_api/ena_handler.py

```diff
@@ -406,13 +401,16 @@ def get_run(
                 )
             )
 
-        if response.status_code == 204:
+        if (
+            response.status_code == requests.codes.no_content
+            or response.status_code == requests.codes.ok
+            and not response.json()
+        ):
             if attempt < 2:
                 attempt += 1
                 sleep(1)
                 logging.warning(
-                    "Error 204 when retrieving run {} in dataPortal {}, "
-                    "retrying {}".format(run_accession, data.get("dataPortal"), attempt)
+                    f"Error 204 when retrieving run {run_accession} in dataPortal {data.get('dataPortal')}, retrying {attempt}"
```

Comment:
:perfection:

Reply:
Sort of. This is to address some data not being available in one data portal (ENA has several) but being available in another. Also, sometimes the response is 404, or 204... or even 200 with an empty (several different cases depending in the kind of data)